### PR TITLE
Improve layout perf

### PR DESCRIPTION
**Pull request by NotWoods**
Switches from [heavily nested layouts to ConstraintLayout](https://android-developers.googleblog.com/2017/08/understanding-performance-benefits-of.html)

I organized the gradle file a little into alphabetical order. Removed unused annotation processor and added def for the glide version (switching one library to the same version as the others).

### CODE REVIEW

**Comment by NotWoods** [comment]
Shrunk the number of changes :)

I'm happy to rebase once #403 is merged in.

I haven't migrated a lot of tokens to Aegis yet so I don't have a good dataset to test perf with. If you know where I can generated a bunch of them I'd be happy to use that. These changes were done following general advice against [RelativeLayout](https://developer.android.com/topic/performance/rendering/optimizing-view-hierarchies) and [deeply nested layouts](https://developer.android.com/training/improving-layouts/optimizing-layout).

**Comment by alexbakker** [comment]
Thanks! Will take a look later today. You can generate a vault with [aegis-tools](https://github.com/alexbakker/aegis-tools), or you can grab one from the testdata folder: https://github.com/beemdevelopment/Aegis/blob/master/testdata/aegis_export_plain.json.

**Comment by alexbakker** [comment]
I was going to take a look at this now, but it looks like all of the unrelated changes are back.

File: app/build.gradle [hunk 1]
```diff
@@ -120,7 +122,7 @@ dependencies {
         transitive = false
     }
     annotationProcessor 'androidx.annotation:annotation:1.1.0'
-    annotationProcessor 'com.github.bumptech.glide:compiler:4.10.0'
+    annotationProcessor "com.github.bumptech.glide:compiler:${glideVersion}"
```
Comment (alexbakker) [inline app/build.gradle:125]: This causes the build to fail.

Comment (NotWoods) [reply app/build.gradle:125]: Oops, looks like that still slipped in. Perf comparison is on my todo list but I've been a bit busy with other work :)

**Comment by alexbakker** [comment]
Just checking in. Do you have still plans to work on this?

**Comment by alexbakker** [comment]
I'm closing this as there hasn't been activity in a while and there've been quite a few conflicting changes since this was first submitted. Feel free to reopen this if you decide to pick this up again.

**Pull request closed without merging**